runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindmillSink.java

```diff
@@ -215,7 +215,13 @@ public long add(WindowedValue<T> data) throws IOException {
               .setMetadata(metadata);
       keyedOutput.addMessages(builder.build());
       keyedOutput.addMessagesIds(id);
-      return (long) key.size() + value.size() + metadata.size() + id.size();
+
+      ByteString offset = ByteString.copyFrom(context.getCurrentRecordOffset());
```

Comment:
would be nice to avoid allocation if no current record since that is common

Reply:
Sounds good, updated to only allocate if non-zero offset.